### ISSUE TITLE
PA causes performance issue when placing blocks

### ISSUE THREAD

**Issue by midspace** [midspace/Space-Engineers-Admin-script-mod#177]
```
Carlosmaid: I wanted to warn you of a possible bug of admin helpers commands. Over a ds, with static ship of 4900 blocks owned by me, when i try to "present" the block to place it, (but without placing it) the client's simspeed drops to .3. When I remove the protection, the simspeed remains constant.
```
![paenabled](https://cloud.githubusercontent.com/assets/6026063/22568082/7a5613ae-e9e6-11e6-8bb9-77d0e0bb3e13.png)
![padisabled](https://cloud.githubusercontent.com/assets/6026063/22568083/7a5cd02c-e9e6-11e6-8f62-6373bb56120d.png)

**Comment by carlosmaid** [comment]
Adding data, safe zone is 50000m lenght and spherical. No others grids on this zone. (empty server). If i change the selected block to a weapon or tools, the simspeed up to 1.0

**Comment by Spcemarine** [comment]
Does the grid have wheels on it?

**Comment by carlosmaid** [comment]
Yes, it has unfinished soldering wheels(cosmetics). If you need,i can try again with the same grid without wheels, and with other large grids of 4900 blocks.

**Comment by Spcemarine** [comment]
Hmm... iirc there was an issue with wheels especially if there are multiple wheels on a single grid. It would be nice if you could check what the simspeed does without the wheels.

**Comment by carlosmaid** [comment]
Confirmed, wheels problem.

A suggestion, within the protection areas only the owner of the grid can plack blocks or grind. It would be better if the people of the faction can do it too, than only the big owner.

**Comment by carlosmaid** [comment]
Extended info: This problem continues even outside the protection zone. When trying to place a block (but not yet placed, not to explain this in English) in a grid with wheels, the simspeed drops abruptly, in the client.

**Comment by carlosmaid** [comment]
the bug is because ProtectedCubePlacer.cs issue or modapi cubeplacer issue?

**Comment by midspace** [comment]
@carlosmaid I've made recent changes to the ProtectedCubePlacer https://github.com/midspace/Space-Engineers-Admin-script-mod/commit/81f4801afdfb49fd563024d9c9ac64068eda36ef which should improve the performance. Are you able to determine if these change have improved your gameplay?

**Comment by carlosmaid** [comment]
Of course yes! happy to help. I retest this and problem solved, same structure, over DS with 16 people online and now all works fine.
I'm going to test structures with dozens of wheels. At the time he noticed that the more wheels had the structure, the greater the lag.
